# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

Backport just the gralloc implementation to 3.0 from the HiHope 3.1 repository.

Signed-off-by: Robert Drab <robert.drab@huawei.com>
Upstream-Status: Inappropriate

diff --git a/hardware/display/BUILD.gn b/hardware/display/BUILD.gn
index f403896..e9adb4f 100644
--- a/hardware/display/BUILD.gn
+++ b/hardware/display/BUILD.gn
@@ -13,13 +13,17 @@
 
 import("//build/ohos.gni")
 
-group("display_group") {
-  deps = [
-    ":display_device",
-    ":display_gfx",
-    ":display_gralloc",
-    ":higbm",
-  ]
+#group("display_group") {
+#  deps = [
+#    ":display_device",
+#    ":display_gfx",
+#    ":display_gralloc",
+#    ":higbm",
+#  ]
+#}
+
+declare_args() {
+  chipset_base_dir = "system"
 }
 
 ohos_static_library("higbm") {
@@ -79,87 +83,87 @@
   part_name = "display_device_driver"
 }
 
-ohos_shared_library("display_gfx") {
-  sources = [ "src/display_gfx/display_gfx.c" ]
-  include_dirs = [
-    "include",
-    "//drivers/peripheral/display/interfaces/include",
-    "//drivers/peripheral/base",
-    "//drivers/framework/include/utils",
-    "//drivers/adapter/uhdf2/adapter/osal/include/",
-    "//utils/native/base/include",
-    "//foundation/graphic/standard/utils/include",
-    "//device/hihope/hardware/rga/include/",
-    "//base/hiviewdfx/interfaces/innerkits/libhilog/include",
-  ]
-  output_name = "display_gfx"
-  cflags = [ "-Wno-macro-redefined" ]
-  deps = [
-    ":display_gralloc",
-    "//device/hihope/hardware/rga:librga",
-    "//utils/native/base:utils",
-  ]
-
-  external_deps = [ "hiviewdfx_hilog_native:libhilog" ]
-
-  install_enable = true
-  install_images = [ chipset_base_dir ]
-  subsystem_name = "hdf"
-  part_name = "display_device_driver"
-}
-
-ohos_shared_library("display_device") {
-  sources = [
-    "src/display_device/drm_connector.cpp",
-    "src/display_device/drm_crtc.cpp",
-    "src/display_device/drm_device.cpp",
-    "src/display_device/drm_display.cpp",
-    "src/display_device/drm_encoder.cpp",
-    "src/display_device/drm_plane.cpp",
-    "src/display_device/drm_vsync_worker.cpp",
-    "src/display_device/hdi_composer.cpp",
-    "src/display_device/hdi_device_interface.cpp",
-    "src/display_device/hdi_display.cpp",
-    "src/display_device/hdi_drm_composition.cpp",
-    "src/display_device/hdi_drm_layer.cpp",
-    "src/display_device/hdi_gfx_composition.cpp",
-    "src/display_device/hdi_layer.cpp",
-    "src/display_device/hdi_netlink_monitor.cpp",
-    "src/display_device/hdi_session.cpp",
-  ]
-  output_name = "display_device"
-  include_dirs = [
-    "src/display_device",
-    "include",
-    "//drivers/peripheral/display/interfaces/include",
-    "//drivers/peripheral/base",
-    "//drivers/framework/include/utils",
-    "//drivers/adapter/uhdf2/osal/include",
-    "//utils/native/base/include",
-    "//foundation/graphic/standard/utils/include",
-    "//foundation/graphic/standard/prebuilts/librarys/drm/include",
-    "//base/hiviewdfx/interfaces/innerkits/libhilog/include",
-  ]
-  deps = [
-    ":display_gfx",
-    ":display_gralloc",
-    "//third_party/libdrm:libdrm",
-    "//utils/native/base:utils",
-  ]
-  cflags = [ "-Wno-unused-function" ]
-  external_deps = [ "hiviewdfx_hilog_native:libhilog" ]
-
-  install_enable = true
-  install_images = [ chipset_base_dir ]
-  subsystem_name = "hdf"
-  part_name = "display_device_driver"
-}
-
-group("display_layer") {
-  deps = [
-    ":display_device",
-    ":display_gfx",
-    ":display_gralloc",
-    ":higbm",
-  ]
-}
+#ohos_shared_library("display_gfx") {
+#  sources = [ "src/display_gfx/display_gfx.c" ]
+#  include_dirs = [
+#    "include",
+#    "//drivers/peripheral/display/interfaces/include",
+#    "//drivers/peripheral/base",
+#    "//drivers/framework/include/utils",
+#    "//drivers/adapter/uhdf2/adapter/osal/include/",
+#    "//utils/native/base/include",
+#    "//foundation/graphic/standard/utils/include",
+#    "//device/hihope/hardware/rga/include/",
+#    "//base/hiviewdfx/interfaces/innerkits/libhilog/include",
+#  ]
+#  output_name = "display_gfx"
+#  cflags = [ "-Wno-macro-redefined" ]
+#  deps = [
+#    ":display_gralloc",
+#    "//device/hihope/hardware/rga:librga",
+#    "//utils/native/base:utils",
+#  ]
+#
+#  external_deps = [ "hiviewdfx_hilog_native:libhilog" ]
+#
+#  install_enable = true
+#  install_images = [ chipset_base_dir ]
+#  subsystem_name = "hdf"
+#  part_name = "display_device_driver"
+#}
+
+#ohos_shared_library("display_device") {
+#  sources = [
+#    "src/display_device/drm_connector.cpp",
+#    "src/display_device/drm_crtc.cpp",
+#    "src/display_device/drm_device.cpp",
+#    "src/display_device/drm_display.cpp",
+#    "src/display_device/drm_encoder.cpp",
+#    "src/display_device/drm_plane.cpp",
+#    "src/display_device/drm_vsync_worker.cpp",
+#    "src/display_device/hdi_composer.cpp",
+#    "src/display_device/hdi_device_interface.cpp",
+#    "src/display_device/hdi_display.cpp",
+#    "src/display_device/hdi_drm_composition.cpp",
+#    "src/display_device/hdi_drm_layer.cpp",
+#    "src/display_device/hdi_gfx_composition.cpp",
+#    "src/display_device/hdi_layer.cpp",
+#    "src/display_device/hdi_netlink_monitor.cpp",
+#    "src/display_device/hdi_session.cpp",
+#  ]
+#  output_name = "display_device"
+#  include_dirs = [
+#    "src/display_device",
+#    "include",
+#    "//drivers/peripheral/display/interfaces/include",
+#    "//drivers/peripheral/base",
+#    "//drivers/framework/include/utils",
+#    "//drivers/adapter/uhdf2/osal/include",
+#    "//utils/native/base/include",
+#    "//foundation/graphic/standard/utils/include",
+#    "//foundation/graphic/standard/prebuilts/librarys/drm/include",
+#    "//base/hiviewdfx/interfaces/innerkits/libhilog/include",
+#  ]
+#  deps = [
+#    ":display_gfx",
+#    ":display_gralloc",
+#    "//third_party/libdrm:libdrm",
+#    "//utils/native/base:utils",
+#  ]
+#  cflags = [ "-Wno-unused-function" ]
+#  external_deps = [ "hiviewdfx_hilog_native:libhilog" ]
+#
+#  install_enable = true
+#  install_images = [ chipset_base_dir ]
+#  subsystem_name = "hdf"
+#  part_name = "display_device_driver"
+#}
+
+#group("display_layer") {
+#  deps = [
+#    ":display_device",
+#    ":display_gfx",
+#    ":display_gralloc",
+#    ":higbm",
+#  ]
+#}
